### ISSUE TITLE
Error executing CGAL Java binding in Ubuntu

### ISSUE THREAD

**Issue by flaviotf** [CGAL/cgal-swig-bindings#104]
I compiled CGAL 4.12 Java bindings for both windows 64 and linux 64 (ubuntu). 
In windows, it works perfectly. However, in Ubuntu I got the following error (same application):
________________________________________________________________________

A fatal error has been detected by the Java Runtime Environment:

 SIGSEGV (0xb) at pc=0x00007fe61f7b50a4, pid=3337, tid=0x00007fe66562e700

JRE version: Java(TM) SE Runtime Environment (8.0_181-b13) (build 1.8.0_181-b13)
Java VM: Java HotSpot(TM) 64-Bit Server VM (25.181-b13 mixed mode linux-amd64 compressed oops)
Problematic frame:
C  [libCGAL_AABB_tree.so+0x700a4]  Java_CGAL_AABB_1tree_CGAL_1AABB_1treeJNI_AABB_1tree_1Triangle_13_1soup_1closest_1point_1and_1primitive_1_1SWIG_10+0x44

Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

An error report file with more information is saved as:
/home/flavio/git/insane/br.ufmg.dees.insane.ui.rich.full/hs_err_pid3337.log
Compiled method (c1)  129040 4365       3       CGAL.AABB_tree.AABB_tree_Triangle_3_soup::closest_point_and_primitive (22 bytes)
 total in heap  [0x00007fe6a9a91a90,0x00007fe6a9a92178] = 1768
 relocation     [0x00007fe6a9a91bb8,0x00007fe6a9a91c18] = 96
 main code      [0x00007fe6a9a91c20,0x00007fe6a9a91f00] = 736
 stub code      [0x00007fe6a9a91f00,0x00007fe6a9a91fa8] = 168
 oops           [0x00007fe6a9a91fa8,0x00007fe6a9a91fb0] = 8
 metadata       [0x00007fe6a9a91fb0,0x00007fe6a9a91fd0] = 32
 scopes data    [0x00007fe6a9a91fd0,0x00007fe6a9a920c0] = 240
 scopes pcs     [0x00007fe6a9a920c0,0x00007fe6a9a92160] = 160
 dependencies   [0x00007fe6a9a92160,0x00007fe6a9a92168] = 8
 nul chk table  [0x00007fe6a9a92168,0x00007fe6a9a92178] = 16
If you would like to submit a bug report, please visit:
  http://bugreport.java.com/bugreport/crash.jsp
The crash happened outside the Java Virtual Machine in native code.
See problematic frame for where to report the bug.

________________________________________________________________________

File hs_err_pid3337.log is attached to this message.

I've already tried to run with oracle-jdk-10, but got the same error.
Any hints on how to solve this error?
[Erro CGAL.txt](https://github.com/CGAL/cgal-swig-bindings/files/2362527/Erro.CGAL.txt)

**Comment by sloriot** [comment]
What is the java code you are trying to run?

**Comment by flaviotf** [comment]
Hi, sloriot.

The error happens during the execution of the following method:

```
  private static void isPointAbove(Vector<Point_3> points, AABB_tree_Triangle_3_soup cutAaBbTree, Vector<Triangle_3> cutTriangles, 
      int[] pointsAbove, Vector<Integer> pointIdsAbove) {
    ListIterator<Point_3> pointsIterator = points.listIterator();
    int objectPointId = 0;
    while (pointsIterator.hasNext()) {
      Point_3 currentPoint = pointsIterator.next();

      Point_and_Integer closestPointOnCutAndIdOfClosestTriangleOfCut = cutAaBbTree.closest_point_and_primitive(currentPoint);
      int primitiveIdOfClosestTriangleOfCut = closestPointOnCutAndIdOfClosestTriangleOfCut.getSecond(); // closest primitive id
      Point_3 closestPointOnCut = closestPointOnCutAndIdOfClosestTriangleOfCut.getFirst();

      Vector_3 currentNormal = new Vector_3();
      Triangle_3 currentTriangle = cutTriangles.get(primitiveIdOfClosestTriangleOfCut);
      calculateNormalOfTriangle(currentTriangle, currentNormal);

      Vector_3 connectionVectorBetweenPointAndReferencePoint = currentPoint.minus(closestPointOnCut);
      double test = currentNormal.times(connectionVectorBetweenPointAndReferencePoint);
      if (test > 0) {
        pointsAbove[objectPointId] = 1;
      } else {
        pointsAbove[objectPointId] = 0;
      }
      if (pointsAbove[objectPointId] == 1) {
        pointIdsAbove.add(objectPointId);
      }
      //  -((currentNormal*connectionVectorBetweenPointAndReferencePoint)==0);
      objectPointId++;
    }
  }
```

One extract of the error log:

```
Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
J 5352  CGAL.AABB_tree.CGAL_AABB_treeJNI.AABB_tree_Triangle_3_soup_closest_point_and_primitive__SWIG_0(JLCGAL/AABB_tree/AABB_tree_Triangle_3_soup;JLCGAL/Kernel/Point_3;)J (0 bytes) @ 0x00007efd34ba132c [0x00007efd34ba12c0+0x000000000000006c]
J 5351 c1 CGAL.AABB_tree.AABB_tree_Triangle_3_soup.closest_point_and_primitive(LCGAL/Kernel/Point_3;)LCGAL/AABB_tree/Point_and_Integer; (22 bytes) @ 0x00007efd2de8faec [0x00007efd2de8f980+0x000000000000016c]
j  br.ufmg.dees.insane.model.gfemmodel.discontinuity.DiscontinuityGeometryHandler3D.isPointAbove(Ljava/util/Vector;LCGAL/AABB_tree/AABB_tree_Triangle_3_soup;Ljava/util/Vector;[ILjava/util/Vector;)V+27
j  br.ufmg.dees.insane.model.gfemmodel.discontinuity.DiscontinuityGeometryHandler3D.init(Ljava/lang/String;Ljava/lang/String;I)V+1863
j  br.ufmg.dees.insane.model.gfemmodel.discontinuity.SurfaceDiscontinuityByGfem.update()V+623
j  br.ufmg.dees.insane.model.gfemmodel.model.GFemModel.evaluateCracksPropagation()V+49
j  br.ufmg.dees.insane.model.gfemmodel.model.GFemModel.update()V+12
j  br.ufmg.dees.insane.assembler.FemAssembler.update()V+4
j  br.ufmg.dees.insane.assembler.GFemAssembler.update()V+1
[hs_err_pid3337.log](https://github.com/CGAL/cgal-swig-bindings/files/2368660/hs_err_pid3337.log)

j  br.ufmg.dees.insane.solution.StaticEquilibriumPath.execute()V+449
j  br.ufmg.dees.insane.ui.rich.solution.SolverClass.run()V+1343
j  java.lang.Thread.run()V+11 java.base@10.0.2
v  ~StubRoutines::call_stub
```

I had attached the wrong file in the previous message. Here goes the correct one.

**Comment by sloriot** [comment]
The following minimal example is working fine for me:
```
import CGAL.Kernel.Point_3;
import CGAL.Kernel.Triangle_3;
import CGAL.AABB_tree.AABB_tree_Triangle_3_soup;
import CGAL.AABB_tree.Point_and_Integer;
import CGAL.Kernel.CGAL_Kernel;

import java.util.LinkedList;
import java.util.Vector;

public class intersect_triangle_soup_with_polyline_aabb{
  public static void main(String[] args)
  {
    double [] t2 = {0,0,0, 2,0,0, 0,2,0};
    double [] t1 = {0,2,0, 2,0,0, 2,2,0};

    double[][] triangles = {t1, t2};

    AABB_tree_Triangle_3_soup tree_tri = new AABB_tree_Triangle_3_soup();
    tree_tri.insert_from_array(triangles);

    Point_3 pt = new Point_3(0.5,0.5,1);
    
    Point_and_Integer res = tree_tri.closest_point_and_primitive(pt);
    System.out.println("closest point is "+res.getFirst());
    System.out.println("closest primitive is "+res.getSecond());
  }
};
```
Could you try update it with you data to see if you can make it crash?

**Comment by flaviotf** [comment]
I ran the code above and it also crashed. 
But it is supposed to crash, as my "triangles" variables is empty!

After a deep analysis, I found that it is another library that I'm using which is not working properly (vtk library) and not correctly filling this variable. 
As stated in my first message, in Windows, it works perfectly, but in linux there is a problem.

I'll keep verifying my code, but it is clear for me now that **_it is not a CGAL problem_**.
Thanks for your help, @sloriot!
